src/tests/DownloadTests.cs

```diff
@@ -20,135 +20,182 @@ public class DownloadTests
         {
-            await TestDownload(purl, targetFilename);
+            await TestDownload(purl, targetFilename, 1);
         }
 
         [DataTestMethod]
         [DataRow("pkg:pypi/bz2file@0.98", "PKG-INFO")]
         public async Task PyPI_Download_Version_Succeeds(string purl, string targetFilename)
         {
-            await TestDownload(purl, targetFilename);
+            await TestDownload(purl, targetFilename, 1);            
         }
 
         [DataTestMethod]
         [DataRow("pkg:ubuntu/zerofree", "zerofree.c")]
         public async Task Ubuntu_Download_Version_Succeeds(string purl, string targetFilename)
         {
-            await TestDownload(purl, targetFilename);
+            await TestDownload(purl, targetFilename, 4);            
         }
 
         [DataTestMethod]
         [DataRow("pkg:vsm/ms-vscode/PowerShell", "extension.vsixmanifest")]
         public async Task VSM_Download_Version_Succeeds(string purl, string targetFilename)
         {
-            await TestDownload(purl, targetFilename);
+            await TestDownload(purl, targetFilename, 1);            
         }
 
-        private async Task TestDownload(string purl, string targetFilename)
+        [DataTestMethod]
+        [DataRow(null, null)]
+        public async Task Null_Test_Download(string purl, string targetFilename)
+        {
+            try
+            {
+                await TestDownload(purl, targetFilename, 1);
+            }
+            catch(FormatException)
+            {
+                return;
+            }
+            Assert.Fail("The right exception did not fire");
+        }
+        private async Task TestDownload(string purl, string targetFilename, int expectedCount)
         {
             string tempDirectoryName = default;
             while (tempDirectoryName == default || File.Exists(tempDirectoryName))
             {
                 tempDirectoryName = Path.GetRandomFileName();
             }
-            
-            var downloadTool = new DownloadTool();
+
             var packageUrl = new PackageURL(purl);
             Assert.IsNotNull(packageUrl);
 
             Directory.CreateDirectory(tempDirectoryName);
+            var downloadTool = new DownloadTool(); // do common initialization
+            PackageDownloader packageDownloader1 = await DownloadPackage(packageUrl, tempDirectoryName);
+            string dir = Directory.GetCurrentDirectory();
+            var fileMatchCount = Directory.EnumerateFiles(tempDirectoryName, targetFilename, SearchOption.AllDirectories).Count();
+            Assert.IsTrue(fileMatchCount > 0);
+
+            // do that again, this should not do anything since the cache already has the package
+            PackageDownloader packageDownloader2 = await DownloadPackage(packageUrl, tempDirectoryName);
+            fileMatchCount = Directory.EnumerateFiles(tempDirectoryName, targetFilename, SearchOption.AllDirectories).Count();
+            Assert.IsTrue(fileMatchCount > 0);
```

Comment:
```suggestion
            Assert.IsTrue(Directory.EnumerateFiles(tempDirectoryName, targetFilename, SearchOption.AllDirectories).Any());
```

It might not matter here, but in general `Any()` is faster than `Count() > 0`.